ui/src/FindBar.tsx

```diff
@@ -92,12 +133,59 @@ export class FindBar extends React.PureComponent<{}, FindBarState> {
     this.dispatchEventToPdfJs("find");
   }
 
-  handleNextButtonClick() {
-    this.dispatchEventToPdfJs("findagain");
+  wrapIndex(index: number, len: number) {
+    return (index % len + len) % len;
   }
 
-  handlePreviousButtonClick() {
-    this.dispatchEventToPdfJs("findagain", true);
+  selectSymbol(id: string, boxId: string) {
+    this.context.setSelectedEntity(id, "symbol");
+    this.context.setSelectedAnnotationId(convertToAnnotationId(id));
+    this.context.setSelectedAnnotationSpanId(boxId);
+  }
+
+  unselectSymbol() {
+    this.context.setSelectedEntity(null, null);
+    this.context.setSelectedAnnotationId(null);
+    this.context.setSelectedAnnotationSpanId(null);
+  }
+
+  moveToNextSymbol(movement: number) {
+    const { matches, mappingToBounds, selectedSymbol } = this.props;
+    const newEntityId = matches[
+      this.wrapIndex(matches.indexOf(selectedSymbol || '') + movement, matches.length)
```

Comment:
Is there a way to structure the `matches` data so that we don't need to look up the index of the selectedSymbol every time we change which symbol is focused? Also, isn't the index of the current symbol stored in the `currentMatch` state property?

Reply:
Yes good point! I'll change that :)